tests/components/test_component_extract_subset.cpp

```diff
@@ -1,8 +1,25 @@
 #include <wmtk_components/extract_subset/internal/extract_subset_2d.hpp>
+#include <wmtk_components/extract_subset/extract_subset.hpp>
 #include <catch2/catch_test_macros.hpp>
+#include "../tools/TriMesh_examples.hpp"
+#include "wmtk/attribute/Attribute.hpp"
+#include "wmtk/Primitive.hpp"
+#include "wmtk/attribute/AttributeHandle.hpp"
+#include "wmtk/Mesh.hpp"
 
-
-TEST_CASE("manual test case", "[components][extract_subset][2D]")
+TEST_CASE("3 neighbors test case", "[components][extract_subset][2D]")
 {
+    wmtk::TriMesh tm;
+    tm = wmtk::tests::three_neighbors(); // start with 4 tri
+    Eigen::Vector<long, 10> V; // init the data vector with any static length, then resize
+    V.resize(tm.capacity( wmtk::PrimitiveType::Face), 1);
+    V.row(0) << 1;
+    V.row(1) << 0; // ont tri is not tagged
+    V.row(2) << 1;
+    V.row(3) << 1;
+    auto tag_handle = wmtk::mesh_utils::set_matrix_attribute(V, "tag", wmtk::PrimitiveType::Face, tm);
```

Comment:
That way of setting values should be only used if you want to assign a value to each vertex at once. It is mostly for the case that you load data from a file. In general, you should follow this example:
https://github.com/wildmeshing/wildmeshing-toolkit/blob/764ca0e341b9ec83f3592dc1f92aebf40caeb6c6/tests/test_accessor.cpp#L54

`register_attribute` can also take a default value for its attribute:
https://github.com/wildmeshing/wildmeshing-toolkit/blob/764ca0e341b9ec83f3592dc1f92aebf40caeb6c6/src/wmtk/Mesh.hpp#L97
You might want to use that to default everything to something signaling invalidness like `-1`.

Reply:
According to the commit history https://github.com/wildmeshing/wildmeshing-toolkit/commit/f1b7140cefd57b767dd338aa4242778ab1c2b8c4, this functionality is update on Oct 5. So in our current branch there is no such interface right now.

Reply:
I did a merge just now.